apps/wallet-mobile/src/features/Exchange/useCases/ShowExchangeResultOrder/ShowExchangeResultOrder.tsx

```diff
@@ -1,45 +1,60 @@
 import {useTheme} from '@yoroi/theme'
 import * as React from 'react'
-import {Image, StyleSheet, TouchableOpacity, View} from 'react-native'
+import {StyleSheet, TouchableOpacity, View} from 'react-native'
 import {SafeAreaView} from 'react-native-safe-area-context'
 
-import banxaLogo from '../../../../assets/img/banxa.png'
 import {Button, Icon, Spacer, Text, useModal} from '../../../../components'
 import {useStatusBar} from '../../../../components/hooks/useStatusBar'
-import {useUnsafeParams} from '../../../../navigation'
-import {useHideBottomTabBar} from '../../../../yoroi-wallets/hooks'
+import {useInitialLink} from '../../../../IntialLinkManagerProvider'
 import {DescribeAction} from '../../common/DescribeAction/DescribeAction'
-import {ExchangeInitRoutes, useNavigateTo} from '../../common/useNavigateTo'
 import {useStrings} from '../../common/useStrings'
+import {BanxaLogo} from '../../illustrations/BanxaLogo'
+import {EncryptusLogo} from '../../illustrations/EncryptusLogo'
 import {WalletAssetImage} from '../../illustrations/WalletAssetImage'
 import {ContentResult} from './ContentResult/ContentResult'
 
-export type ParamsResult = {
-  coin: string
-  coinAmount: number
-  fiat: number
-  fiatAmount: number
-}
-
-export const ShowExchangeResultOrder = ({variant}: {variant?: 'noInfo'}) => {
+export const ShowExchangeResultOrder = ({onClose}: {onClose: () => void}) => {
   const strings = useStrings()
-  useHideBottomTabBar()
+  const {setInitialUrl, initialUrl} = useInitialLink()
   useStatusBar()
-  const navigateTo = useNavigateTo()
   const styles = useStyles()
-  const params = useUnsafeParams<ExchangeInitRoutes['rampOnOff']>()
-  const {coin, coinAmount, fiat, fiatAmount} = params ?? {}
+
+  const urlSearchParams = initialUrl !== null ? new URLSearchParams(initialUrl) : null
+  const params =
+    urlSearchParams !== null
+      ? {
+          coin: urlSearchParams.get('coin'),
+          coinAmount: urlSearchParams.get('coinAmount'),
+          fiat: urlSearchParams.get('fiat'),
+          fiatAmount: urlSearchParams.get('fiatAmount'),
+          provider: urlSearchParams.get('provider'),
+          status: urlSearchParams.get('status'),
+        }
+      : null
+
+  const hasMinimunParams =
+    params !== null &&
+    params.coin !== null &&
+    params.coinAmount !== null &&
+    params.fiat !== null &&
+    params.fiatAmount !== null
 
   const {openModal} = useModal()
 
   const handleDirectTransaction = () => {
-    navigateTo.rampOnOffOpenOrder()
+    setInitialUrl(null)
+    onClose()
   }
 
   const handlePressDescribe = () => {
     openModal(strings.buySellCrypto, <DescribeAction />)
   }
 
+  const isEncryptus = params?.provider != null && params?.provider === 'encryptus'
+
+  const Logo = isEncryptus ? EncryptusLogo : BanxaLogo
+  const label = isEncryptus ? 'Encryptus' : 'Banxa'
```

Comment:
not the best solution. open to ideas. here we need to consider banxa cant attach custom params. i will ask them anyway

Reply:
it is not possible to use Exchange context here because this screen opens in the wallet selection screen (and tx history)